src/app/view/perspectives/post-it/post-it-perspective.component.ts

```diff
@@ -110,22 +115,18 @@ export class PostItPerspectiveComponent implements OnInit, OnDestroy {
       this.workspace = workspace;
       this.query = query;
 
-      this.initializeLayout();
       this.fetchPostIts();
+      this.initializeLayout();
+      this.setCurrentCollection();
     });
   }
 
   private initializeLayout(): void {
-    if (this.layout) {
-      this.refresh();
-      return;
-    }
+      this.layout = new PostItLayout('.post-it-document-layout', new PostItLayoutConfig(), this.zone);
+  }
 
-    this.layout = new PostItLayout({
-      container: '.post-it-document-layout',
-      item: '.layout-item',
-      gutter: 10
-    });
+  private setCurrentCollection() {
+    this.currentCollection = (this.query && this.query.collectionCodes) ? this.query.collectionCodes[0] : null;
```

Comment:
Would not this reintroduce the issue that was there before? What will happen if there are documents from multiple collections shown in this perspective?

Reply:
If there are multiple the add button will not show at all.

This in fact introduced a new issue. After loading the last page of documents. The newly loaded pages are empty.. so their array of CollectionCodes was empty. Making `currentCollection === null`